src/client/metadataApiDeploy.ts

```diff
@@ -313,4 +322,31 @@ export class MetadataApiDeploy extends MetadataTransfer<MetadataApiDeployStatus,
   protected async post(result: MetadataApiDeployStatus): Promise<DeployResult> {
     return new DeployResult(result, this.components);
   }
+
+  private async getZipBuffer(): Promise<Buffer> | undefined {
```

Comment:
When does this return undefined?  If it does return undefined it would still need to be `Promise<void>` right?  Seems like either `Promise<Buffer>` or `Promise<Optional<Buffer>>` is the return type.

Reply:
I'm just gonna throw at the bottom